### DISCUSSION TITLE
Integration of ICA and ICQ modules on Agoric

### DISCUSSION THREAD

**Discussion by Jorge-Lopes** [Agoric/agoric-sdk#8540] [General]
Greetings,

This discussion is a follow-up to the [last office hours](https://github.com/Agoric/agoric-sdk/discussions/8532) with @dckc .
It serves the purpose of tracking the status of the IBC modules integration on Agoric.
More specifically,  the Controller submodule of [Interchain Accounts](https://github.com/cosmos/ibc/blob/main/spec/app/ics-027-interchain-accounts/README.md) and the [Interchain Queries](https://github.com/strangelove-ventures/async-icq).

As mentioned by @rowgraus, this task is already being addressed by Agoric. Which made us excited to see the new coming features.

If there is any way we could contribute, we would be glad to help.

Thank you and best regards.

**Comment by dckc** [comment]
Note **GOAL02. Prepare for cross-chain capability launch** in today's post by Rowland:

 - [Q4 2023 \- Product & Eng Goals \- Platform & BLDers \- Agoric Community Forum](https://community.agoric.com/t/q4-2023-product-eng-goals/543)

cc @michaelfig

**Comment by dckc** [comment]
> It serves the purpose of tracking the status of the IBC modules integration on Agoric.

This is certainly a place to discuss progress on these, but if there's something specific that should be tracked, it's usually worth capturing in an issue. We're currently tracking some related stuff:

 - https://github.com/Agoric/agoric-sdk/issues/8223
 - https://github.com/Agoric/agoric-sdk/issues/8224
 - ... issues labelled [agoric-cosmos](https://github.com/Agoric/agoric-sdk/labels/agoric-cosmos)

If you have something you tried to do with interchain accounts and you can detail the steps that led up to a failure, that might make for a handy issue.

**Reply by Jorge-Lopes** [reply]
Thank you very much for your replies @dckc 

We are currently in the planning stage and haven't initiated any development in this regard. 
Our objective is to create an ICA controlled by an Agoric contract. 
The ICQ module will serve as a valuable tool for querying the ICA balance.

**Reply by michaelfig** [reply]
If you want to enable a JS contract to control an ICA account on another chain, there is [an Agoric Component for that](https://components.agoric.com/smart-contracts/cross-chain/cosmos-hub-ica).

Note that we haven't deployed a contract using that component on mainnet yet, because it relies on the Network and IBC vats that need to be made upgradable first.
